app/overrides/spree/order_deface.rb

```diff
@@ -0,0 +1,28 @@
+script = '
+<% intent_secrets = @order.payments.map do |payment|
+  next unless payment.intent_client_key
+  {
+    intent_key: payment.intent_client_key,
+    pk_key: payment.payment_method.preferred_publishable_key
+  }
+end.last %>
+
+<% if intent_secrets %>
```

Comment:
```suggestion
<% if intent_secrets && order_just_completed?(@order) %>
```

@chhga shouldn't this fire only once?